### PR TITLE
Add trail data to the returned structure

### PR DESCRIPTION
**Pull request by MarquisdeGeek**
Having trail data available, but not returned, seems like a wasted opportunity.

### CODE REVIEW

**Review by derhuerst — COMMENTED** [review]
Let's parse the fields in `d.trail` like those in `radar()`: https://github.com/derhuerst/flightradar24-client/blob/49e2953b9ca7c61ead31cf95688b5376c2a6100d/lib/radar.js#L86-L90

**Review by derhuerst — COMMENTED** [review]
Please also adapt `test.js`.

File: lib/flight.js [hunk 1]
```diff
@@ -48,6 +48,15 @@ const fetchFlight = async (flight) => {
...
 	if (delay) delay = parseInt(delay)
 
+	let trail = d.trail?.map((t) => ({
+		latitude: t.lat ?? null,
+		longitude: t.lng ?? null,
+		altitude: t.alt ?? null,
+		bearing: t.hd ?? null, // in degrees
+		speed: t.spd ?? null,
+		timestamp: t.ts ?? null,
+	}));
```
Comment (derhuerst) [inline lib/flight.js:58]:
```suggestion
	})) ?? null;
```

File: test.js [hunk 2]
```diff
@@ -122,6 +122,10 @@ test('fetchFlight', async (t) => {
 			t.equal(typeof flight.arrivalGate, 'string')
 		}
 		if (flight.delay !== null) t.equal(typeof flight.delay, 'number')
+
+		if (flight.trail) {
```
Comment (derhuerst) [inline test.js]:
```suggestion
		if (flight.trail !== null) {
```

File: test.js [hunk 3]
```diff
@@ -122,6 +122,10 @@ test('fetchFlight', async (t) => {
...
+			t.equal(typeof flight.trail, 'object')
```
Comment (derhuerst) [inline test.js]: Please check the individual fields. This assertion doesn't help much.

File: lib/flight.js (hunk 1 above)
Comment (MarquisdeGeek) [reply lib/flight.js:58]: My intention was that a lack of trail data return [], as the covered case in line 98:

rail: trail || []

This was so the results could be handled with a trail.forEach() without first checking for null.

Do you consider null to be a better choice?

File: test.js (hunk 2 above)
Comment (MarquisdeGeek) [reply test.js]: Will await comment on above issue, before resolving this in the appropriate manner.

File: lib/flight.js [hunk 4]
```diff
@@ -86,9 +95,9 @@ const fetchFlight = async (flight) => {
 		arrivalTerminal: dest.info?.terminal ?? null,
 		arrivalGate: dest.info?.gate ?? null,
 		delay, // in seconds
+		trail: trail || []
```
Comment (derhuerst) [inline lib/flight.js]:
```suggestion
		trail: trail || [],
```

**Comment by derhuerst** [comment]
I have squashed your commits into 42017d8, and merged them.

Thanks! Do you want to be [mentioned as a contributor in `package.json`](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#people-fields-author-contributors)? If you do, how? As `MarquisdeGeek  <marquisdegeek@gmail.com>`, or just `MarquisdeGeek`?

**Comment by MarquisdeGeek** [comment]
I don't think I've done enough (yet?) to deserve it. Maybe next time!

**Pull request merged**